packages/babel-parser/src/parser/statement.js

```diff
@@ -1648,7 +1648,19 @@ export default class StatementParser extends ExpressionParser {
   }
 
   parseClassSuper(node: N.Class): void {
-    node.superClass = this.eat(tt._extends) ? this.parseExprSubscripts() : null;
+    const superClass = this.eat(tt._extends)
+      ? this.parseExprSubscripts()
+      : null;
+    if (superClass !== null) {
```

Comment:
I had to use 2 if stmts instead of 1 because Flow can't recognize that `superClass !== null` if I do `if (superClass !== null && ...` (I think).